app/controllers/exercises_controller.rb

```diff
@@ -10,4 +10,10 @@ def show
     @exercise = Exercise.find(params[:id])
     UserExercise.find_or_create_by(user_id: current_user.id, exercise_id: @exercise.id)
   end
+
+  def update
+    user_exercise = UserExercise.find_by(user_id: current_user.id, exercise_id: params[:id])
+    user_exercise.update_attribute(:status, "finished")
```

Comment:
So what I was saying is, the update action in Rails is reserved for updating the whole object. This action is reserved for updating the resource with any attribute that might be passed in the `object_params` hash. Here, the logic just sets the user exercise status to finished, so a more appropriate action name is `update_status`.

We're hard coding a value it will get updated to, but if somewhere within our app, we need to update a user exercises with a status that comes in as an attribute from the params hash, then this action would be more suitable.